src/tools/run-make-support/src/fs.rs

```diff
@@ -0,0 +1,86 @@
+use std::fs;
+use std::path::Path;
+
+/// A safe wrapper around `std::fs::remove_file` that prints the file path if an operation fails.
+pub fn remove_file<P: AsRef<Path> + std::fmt::Debug>(path: P) {
```

Comment:
Question: does this *need* `+ std::fmt::Debug`? AFAIK `Path` already implements debug? Same with all the other bounds that have `+ std::fmt::Debug`.

Reply:
Not sure, but I think the `AsRef` might make it lose the `Debug` implementation? Anyhow, the `{:?}` sprinkled across `run-make-support` return an error if we try to pass the `Path` into them with `as_ref()`.

I have replaced `+ std::fmt::Debug` with `.display()`, as it is what is already being done in `run-make-support`.